bindings/python/edlib.pyx

```diff
@@ -152,3 +154,98 @@ def align(query, target, mode="NW", task="distance", k=-1, additionalEqualities=
+            the faster is calculation. Set to -1 (default) to have no limit on edit distance.
+    @param {list} additionalEqualities  Optional.
+            List of pairs of characters or hashable objects, where each pair defines two values as equal.
+            This way you can extend edlib's definition of equality (which is that each character is equal only
+            to itself).
+            This can be useful e.g. when you want edlib to be case insensitive, or if you want certain
+            characters to act as a wildcards.
+            Set to None (default) if you do not want to extend edlib's default equality definition.
+    @return Dictionary with following fields:
+            {string} cigar  Cigar is a standard format for alignment path.
+                Here we are using extended cigar format, which uses following symbols:
+                Match: '=', Insertion to reference: 'I', Deletion from reference: 'D', Mismatch: 'X'.
+                e.g. cigar of "5=1X1=1I" means "5 matches, 1 mismatch, 1 match, 1 insertion (to reference)".
+            {string} NICE  NICE is the indented pairwise alignment. 
+            {string} query_aln  query_aln is the aligned query sequence from the input 'query'
+            {string} match_aln  match_aln is the pairwise alignment from the parsed cigar
+                The symbol '|' represents a read match '='
+                The symbol '.' represents a read mismtach 'X'
+                The space ' ' represents an insertion 'I' or deletion 'D'
+            {string} ref_aln  ref_aln is the aligned reference sequence from the input 'reference'                
+    """
+
+    ## 'path' - find edit distance, start and end locations and alignment path. (Default)
+
+    result = align(query=query, target=reference, mode=mode, task="path", k=k, additionalEqualities=additionalEqualities)
+
+    ## access locations, a list of a set [(start, end)], from yech1990
+    ## ref_pos = locations[0][0]
+    ref_pos = result["locations"][0][0]
+    if ref_pos == None:
+        ref_pos = 0
+    query_pos = 0 ## 0-indexed
+    ref_aln = match_aln = query_aln = ""
+
+    cigar = result["cigar"]
+
+    ## cigar parsing, from yech1990
+    for step, code in re.findall("(\d+)(\D)", cigar):
+        step = int(step)
+        if code == "=":
+            ref_aln += reference[ref_pos : ref_pos + step]
+            ref_pos += step
+            query_aln += query[query_pos : query_pos + step]
+            query_pos += step
+            match_aln += "|" * step
+        elif code == "X":
+            ref_aln += reference[ref_pos : ref_pos + step]
+            ref_pos += step
+            query_aln += query[query_pos : query_pos + step]
+            query_pos += step
+            match_aln += "." * step
+        elif code == "D":
+            ref_aln += reference[ref_pos : ref_pos + step]
+            ref_pos += step
+            query_aln += " " * step
```

Comment:
You suggested we go with "-" instead of " " -> sounds reasonable, it is easier to read length of gap!
However, one thing to keep in mind is that edlib is not used only by bioinformaticians but also by other people, who sometimes input whole alphabet in it. So question is, which character makes sense? What if "-" is part of the query or target? So we could consider having that as an optional parameter, and default would be "-".

Reply:
> So we could consider having that as an optional parameter, and default would be "-".

That sounds like a sensible approach. 

Would you recommend some sort of check to make sure users do not use a parameter which isn't suitable? Perhaps "check if character" is too rigid.

Reply:
Another question:

What would you name this parameter? `gap`?

Reply:
Hmmm, we could check for them if parameter is not suitable, for example by checking if it is contained in query or target and rejecting it if so. However, that sounds like too much work to me / thinking too much ahead, maybe I would rather just have them pick it for now and leave it to them to figure out what they want to use. If it becomes a problem, we can always add this functionality later.

Naming: `gap` is good! Slightly more informative could be `gapSymbol`, although `gap`is also acceptable for me.

Reply:
> If it becomes a problem, we can always add this functionality later.

Agreed

> Naming: gap is good! Slightly more informative could be gapSymbol, although gapis also acceptable for me.

I like `gapSymbol` actually